File: missing_number/easiest.js
// Run Time 116 ms, Memory Usage 37.2 mb

function missingNumber(arr){
    // sort array in order
    arr.sort(function(a, b){ return a - b })
    
    // iterate through array
    for (let i = 0; i < arr.length; i++){
        // since array should start at 0, the index should match the number
        if (arr[i] !== i){
            return i
        } else if (i === arr.length - 1){
            // if every number corresponds, we need to return the next value in the sequence
            return arr.length
        }
    }
}

File: missing_number/easiest.rb
# Runtime 720 ms, Memory Usage 11.7 mb

def missing_number(nums)
    # sort the array to get an order
    sortNums = nums.sort
    sortNums.each do |num|
        # since array should be starting with 0, the index should match the number
        if num != sortNums.find_index(num)
            return sortNums.find_index(num)
        # if we get to the end and have all numbers, return the next number in the sequence
        elsif num === sortNums.length - 1
            return sortNums.length
        end
    end
end